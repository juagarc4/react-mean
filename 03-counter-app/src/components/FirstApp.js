// import React, { Fragment } from 'react';
import React from 'react';
import PropTypes from 'prop-types';

const FirstApp = ({ greeting, subtitle }) => {


    // const greeting = 'Hola mundo';
    // const user = {
    //     name: 'Raul',
    //     age: 22
    // }
    //<> === <Fragment ><Fragment />
    console.log();
    return (
        <>
            <h1>{ greeting }</h1>
            {/*<pre>{ JSON.stringify(user, null, 3) }</pre>*/}
            <p>{ subtitle }</p>
        </>
    );
}

FirstApp.propTypes = {
    greeting: PropTypes.string.isRequired,
}
FirstApp.defaultProps = {
    subtitle : 'Soy un subtitulo',
}
export default FirstApp;