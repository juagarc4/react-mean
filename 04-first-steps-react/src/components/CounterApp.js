import React, { useState } from 'react';
import PropTypes from 'prop-types';

const CounterApp = ({ value = 10 }) => {

    const [ counter, setCounter ] = useState(value);

    const handleClick = () => {
         setCounter( counter + 1 );
        // Alternative version
        // setCounter( (c) => c + 1 );
    }

    const handleSub = () => {
         // setCounter( counter - 1 );
        // Alternative version
        setCounter( (c) => c - 1 );
    }

    const handleReset = () => setCounter( value );

    return (
        <>
            <h1>CounterApp</h1>
            <h2> { counter } </h2>
            <button onClick={ handleClick }>+1</button>
            <button onClick={ handleReset }>Reset</button>
            <button onClick={ handleSub }>-1</button>
        </>
    )
}
CounterApp.propTypes = {
    value: PropTypes.number,
}
export default CounterApp;