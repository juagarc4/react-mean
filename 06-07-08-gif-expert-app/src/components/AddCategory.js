import React, { useState } from 'react'
import PropTypes from 'prop-types'

export const AddCategory = ({ setCategories }) => {
  const [inputValue, setInputValue] = useState('')

  const handleInputChange = (e) => {
    setInputValue(e.target.value)
    console.log('handleInputChange llamado')
  }

  const handleSubmit = (e) => {
    e.preventDefault()
    if (inputValue.trim().length > 0) {
      setCategories((cats) => [inputValue, ...cats])
      setInputValue('')
      console.log('handleSubmit llamado')
    }
  }

  return (
    <form onSubmit={handleSubmit}>
      <input type="text" value={inputValue} onChange={handleInputChange} />
    </form>
  )
}

AddCategory.propTypes = {
  setCategories: PropTypes.func.isRequired,
}
