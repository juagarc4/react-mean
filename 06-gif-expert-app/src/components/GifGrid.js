import React from 'react'

export const GifGrid = ({ category }) => {
  const getGifs = async () => {
    const url = 'https://api.giphy.com/v1/gifs/search?q=Dragon+Ball&limit=10&api_key=apRdaKcwOzqVBsDIpZE2MzSssnZXNexp'
    const resp = await fetch(url)
    const { data } = await resp.json()
    const gifs = data.map((img) => {
      return {
        id: img.id,
        title: img.title,
        url: img.images?.downsized_medium.url,
      }
    })
  }
  getGifs()
  return (
    <div>
      <h3>{category}</h3>
    </div>
  )
}
