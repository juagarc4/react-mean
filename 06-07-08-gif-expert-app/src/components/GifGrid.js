import React from 'react'
import { GifGridItem } from 'components/GifGridItem'
import { useFetchGifs } from 'hooks/useFetchGifs'
import PropTypes from 'prop-types'

export const GifGrid = ({ category }) => {
  const { gifs, loading } = useFetchGifs(category)

  return (
    <>
      <h3>{category}</h3>
      {loading && (
        <p className="animate__animated animate__flash">Loading...</p>
      )}
      <div className="card-grid">
        {gifs && gifs.map((gif) => <GifGridItem key={gif.id} {...gif} />)}
      </div>
    </>
  )
}

GifGrid.propTypes = {
  category: PropTypes.string.isRequired,
}
