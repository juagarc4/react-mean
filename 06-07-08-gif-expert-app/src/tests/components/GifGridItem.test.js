import React from 'react'
import { GifGridItem } from 'components/GifGridItem'
import { shallow } from 'enzyme'
const title = 'A title for a test image'
const url = 'https://media.giphy.com/media/3orieKKmYyvUdR3RkY/giphy.gif'
const wrapper = shallow(<GifGridItem title={title} url={url} />)

describe('Tests in <GifGridItem />', () => {
  test('Should show the component correctly', () => {
    expect(wrapper).toMatchSnapshot()
  })

  test('Should have a p element with the title', () => {
    const p = wrapper.find('p')
    expect(p.text().trim()).toBe(title)
  })

  test('Should have an img element with the url and alt', () => {
    const img = wrapper.find('img')
    expect(img.props().src).toBe(url)
    expect(img.props().alt).toBe(title)
  })

  test('Should have the class animate__fadeIn', () => {
    const div = wrapper.find('div')
    expect(div.props().className).toContain('animate__fadeIn')
  })
})
