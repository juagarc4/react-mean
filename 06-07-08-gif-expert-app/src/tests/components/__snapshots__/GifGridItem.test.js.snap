// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in <GifGridItem /> Should show the component correctly 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="A title for a test image"
    src="https://media.giphy.com/media/3orieKKmYyvUdR3RkY/giphy.gif"
  />
  <p>
    A title for a test image
  </p>
</div>
`;
